### PR TITLE
Tidy up test targets

### PR DESCRIPTION
**Pull request by froesef**
This PR adds the connectors and pkg directly to the `test` target of the main make file.

### CODE REVIEW

File: pkg/helm/helm_test.go [hunk 1]
```diff
@@ -79,6 +79,11 @@ func TestHelmCache(t *testing.T) {
 }
 
 func TestHelmRegistry(t *testing.T) {
+	// Test should only run as integration test if registry is available
+	if host, isSet := os.LookupEnv("DOCKER_HOSTNAME"); !isSet || host != "localhost:5000" {
```
Comment (shlomitk1) [inline pkg/helm/helm_test.go:83]: Not sure about this. Theoretically, I can run integration tests from any environment.

File: pkg/helm/helm_test.go [hunk 2]
```diff
@@ -109,6 +114,10 @@ func TestHelmRegistry(t *testing.T) {
 }
 
 func TestHelmRelease(t *testing.T) {
+	// Test should only run as integration test if registry is available
+	if host, isSet := os.LookupEnv("DOCKER_HOSTNAME"); !isSet || host != "localhost:5000" {
```
Comment (shlomitk1) [inline pkg/helm/helm_test.go:118]: Same comment

File: pkg/vault/vault_interface_test.go [hunk 3]
```diff
@@ -21,7 +22,7 @@ func Log(t *testing.T, label string, err error) {
 
 func TestCredentialManagerInterface(t *testing.T) {
 	var err error
-	os.Setenv("RUN_WITHOUT_VAULT", "0")
+	os.Setenv("RUN_WITHOUT_VAULT", "1")
```
Comment (shlomitk1) [inline pkg/vault/vault_interface_test.go:25]: We want to test real vault connection, not the dummy interface.

File: pkg/helm/helm_test.go (hunk 1 above)
Comment (froesef) [reply pkg/helm/helm_test.go:83]: True. So how about dropping the host check and only skipping the test if the `DOCKER_HOSTNAME` environment variable is not set?

File: pkg/vault/vault_interface_test.go (hunk 3 above)
Comment (froesef) [reply pkg/vault/vault_interface_test.go:25]: Yes. Is there a real vault instance we can run against? E.g. in the integration test one could run against vault and in the unit test against the dummy?

File: pkg/helm/helm_test.go (hunk 1 above)
Comment (shlomitk1) [reply pkg/helm/helm_test.go]: Agree.

File: pkg/vault/vault_interface_test.go (hunk 3 above)
Comment (shlomitk1) [reply pkg/vault/vault_interface_test.go:26]: Both helm and vault packages are tested in integration tests only, not in unit tests, if I am not mistaken.

**Comment by roee88** [comment]
Thanks for this PR, you're touching a painful issue in the codebase.

Some general comments:

1. This project is a tree of makefiles, I keep getting lost and it keeps introducing duplications and errors. I strongly recommend to stop using this pattern or at least limit its use to minimum. For example, *none* of the makefiles in pkg should exist.
2. `go test ./...` runs recursively. You can't run it in pkg and then separately in things under pkg, it's duplicate work. It's actually more than duplicate work, it will run tests that require vault to be setup but those should only run through the test script in pkg/vault. Which brings me to the next item
3. We should try to minimize situations that *require* setting up things outside of the test suites (good example: motion controller tests; bad example: pkg/vault test). For example for vault testing see https://stackoverflow.com/a/57773764.

**Comment by shlomitk1** [comment]
> 
> 
> Otherwise, a clear separation of integration tests is needed.

We don't really want to test vault, only that the vault deployment in the control plane is working as expected. Same goes for helm. 
This is another problem: how to run tests that do require a full setup? Running same tests both in unit tests and as end-to-end tests in an existing cluster has its issues - hence special casing whether EXISTING_CONTROLLER/CLUSTER is set or not.

**Comment by roee88** [comment]
> > Otherwise, a clear separation of integration tests is needed.
> 
> We don't really want to test vault, only that the vault deployment in the control plane is working as expected. Same goes for helm.

I don't understand this statement. The tests in both cases test the client code in those packages. It doesn't and shouldn't test that vault is installed correctly as far as I can tell.

**Comment by shlomitk1** [comment]
> Agree. My question is do we want to do this special casing for every test or just create completely septate integrations test (e.g. in /test).

I vote for completely separate integration tests.

File: pkg/vault/vault_interface_test.go (hunk 3 above)
Comment (froesef) [reply pkg/vault/vault_interface_test.go:26]: They are currently not tested at all. And for the integration tests they need an exposed port that currently is not exposed in the integration tests.

**Comment by froesef** [comment]
I think what the vault test should test is if our programs can run against a vault interface. (And as @roee88 mentioned there are utilities from vault that are designed to be used in such unit test situations. Thus there is no need for an external vault setup (or integration tests))

The same is actually also valid for out helm and integration tests. There we want to test against a service that exposes a helm api or a Kubernetes api (e.g. when posting Blueprints, etc in the integration tests) But as there are no good in memory alternatives for the two that simulate the appropriate behaviour (e.g. also run pods etc) we're running these as integration tests against a kind environment.

I think we should keep the tests in the packages where their code is if they are specialised e.g. on just motion, app, helm etc. 

If we're really testing a full fledged custom setup where we're writing test values into vault, constructing an M4dApplication, installing modules etc in order to e.g. run a real copy and afterwards check S3 if it was really executed this can be under the `test` folder.

Lastly I agree with @roee88 that we should at some point just have one `go test ./...` in the root.

File: Makefile [hunk 4]
```diff
@@ -17,7 +17,8 @@ build:
 
 .PHONY: test
 test:
-	$(MAKE) -C pkg/policy-compiler test
+	$(MAKE) -C connectors test
+	$(MAKE) -C pkg test
```
Comment (roee88) [inline Makefile:21]: I suggest to remove this (and the makefile in pkg) from this PR. It will break tests unless ugly workarounds are added.

Comment (roee88) [reply Makefile:21]: Related to my other comment about this running recursively also the vault tests.

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -9,8 +9,6 @@ on:
       - master
 
 env:
-  DOCKER_HOSTNAME: ghcr.io
```
Comment (roee88) [inline .github/workflows/build.yml:12]: You can remove these from the verify job too (I think)

File: manager/Makefile [hunk 6]
```diff
@@ -15,6 +15,8 @@ include $(ROOT_DIR)/hack/make-rules/verify.mk
 config/default/kustomization.yaml: config/default/kustomization.yaml.in
 	cp $< $@
 
+pre-test: generate manifests $(TOOLBIN)/etcd $(TOOLBIN)/kube-apiserver
```
Comment (roee88) [inline manager/Makefile:18]: Minor: you can use this in `test` below to avoid repeating.

**Review by roee88 — APPROVED** [review]

**Review by roee88 — APPROVED** [review]

**Pull request merged**